### PR TITLE
Fix prog button handling

### PR DESCRIPTION
**Pull request by rueckix**
Addressing https://github.com/thelsing/knx/discussions/206

A button press is now detected as _valid_ if it is longer than `PROG_BTN_PRESS_MIN_MILLIS` and shorter than `PROG_BTN_PRESS_MAX_MILLIS`. Each with reasonable defaults of 50ms and 500ms.

This removes the need for setting edge detection to RISING of FALLING. The interrupt will simply trigger on CHANGE.

### CODE REVIEW

File: src/knx_facade.h [hunk 1]
```diff
@@ -423,7 +406,7 @@ template <class P, class B> class KnxFacade : private SaveRestore
     ProgLedOffCallback _progLedOffCallback = 0;
     uint32_t _ledPinActiveOn = LOW;
     uint32_t _ledPin = LED_BUILTIN;
-    uint32_t _buttonPinInterruptOn = RISING;
+    uint32_t _buttonPinInterruptOn = CHANGE;
```
Comment (thelsing) [inline src/knx_facade.h:409]: Just get rid of that field and use CHANGE directly. Or is there a need to keep it?

File: src/stm32_platform.cpp [hunk 2]
```diff
@@ -32,16 +32,21 @@ void Stm32Platform::restart()
 
 uint8_t * Stm32Platform::getEepromBuffer(uint16_t size)
 {
-    if (size > E2END + 1)
```
Comment (thelsing) [inline src/stm32_platform.cpp:35]: Strictly speaking this change should be in a different PR. But since it's only a small change its ok.

**Review by thelsing — APPROVED** [review]

File: src/stm32_platform.cpp (hunk 2 above)
Comment (rueckix) [reply src/stm32_platform.cpp:35]: That's not a change. It must have happened as as re-sync a branch with thelsing:master.  Sorry for the slight mess.

The code is identical. 

master: https://github.com/thelsing/knx/blob/master/src/stm32_platform.cpp
patch: https://github.com/rueckix/knx/blob/patch-progbutton/src/stm32_platform.cpp

File: src/knx_facade.h (hunk 1 above)
Comment (rueckix) [reply src/knx_facade.h:409]: No need to keep it. I'll kill it.

**Comment by thelsing** [comment]
Merged. Thanks.
